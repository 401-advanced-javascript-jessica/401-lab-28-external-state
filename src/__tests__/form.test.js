import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../components/Form/form.js';

describe("<Form />", () => {
    it('form is rendered at application start', () => {
        let app = shallow(<Form />);
        expect(app.find('form').exists()).toBe(true);
        expect(app.find('input').exists()).toBe(true);
        expect(app.find('button').exists()).toBe(true);
    });

    it('renders properly', ()=> {
        const tree = renderer.create(<Form />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
