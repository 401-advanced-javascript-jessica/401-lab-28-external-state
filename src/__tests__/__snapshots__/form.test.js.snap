// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    type="submit"
  >
     Update | 
  </button>
</form>
`;
